pymemcache/client.py

```diff
@@ -999,6 +1009,256 @@ def __delitem__(self, key):
         self.delete(key, noreply=True)
 
 
+class ShardingClient(object):
```

Comment:
The client.py file is getting rather large. Let's move ShardingClient to a new file called sharding_client.py. I'll do the same for PooledClient in a separate commit.